### ISSUE TITLE
can't use some apps with touchscreen

### ISSUE THREAD

**Issue by gmankab** [YaLTeR/niri#889]
### System Information

* niri version: niri 0.1.10-1 (75c7911)

* Distro: Fedora 41 workstation

* GPU: Qualcomm Turnip Adreno (TM) 640 [Integrated]

* CPU: Qualcomm Snapdragon 860 (7 nm) (sm8150 (8) @ 2.96 GHz)

how to reproduce:

- get device with touchscreen (xiaomi pad 5 in my case)
- install fedora 41
- install niri from corp

problematic apps:
- telegram desktop from flathub
- libre office from flathub

in telegram desktop i can open and scroll chats, but can't send emoji, can't select text, and can't close app using touchscreen while in niri session

but this all works in niri with mouse, also works in gnome wayland session with both mouse and touchscreen

working apps:
- kde dolphin
- kde kate
- gnome nautilus
- gnome calculator
- gnome web
- firefox

in this apps all features are working with touchsceen

**Comment by YaLTeR** [comment]
Are you sure that on GNOME your Telegram isn't running as an X11 client? You can check with `xeyes`.

**Comment by gmankab** [comment]
@YaLTeR

i restricted telegram access to x11 with flatseal, still works in gnome, and still problems in niri

```shell
> xlsclients
xiaomi-nabu  gsd-xsettings
xiaomi-nabu  ibus-x11
xiaomi-nabu  mutter-x11-frames
```

xlsclients not shows telegram, so telegram uses wayland

**Comment by YaLTeR** [comment]
Thanks, I reproduced it. Seems like the difference is that Smithay sends a wl_touch frame event after wl_touch up. I opened an issue there: https://github.com/Smithay/smithay/issues/1625

**Comment by YaLTeR** [comment]
After some more debugging, seems that this is actually a Telegram/Qt bug with fractional scaling. With scale=1 on both of my screens, touch works fine in Telegram on niri.

**Comment by gmankab** [comment]
@YaLTeR, i found out that issue related not only to the scaling, but also to screen rotation

on xiaomi pad (fedora), gnome by default runs in landscape mode, but niri runs in portrait mode

in gnome, touchscreen in telegram works when in landscape mode (default one), and does not works when i am in portrait mode (switched in settings)

in niri everything is reverted, touchscreen works in portrait mode (default one), and does not works in portrait mode (set `transform "270"` in config)

with keyboard, xiaomi pad 5 is only usable in portrait mode, unfortunately

on poco f1 (postmarket os, phosh) tocuhscreen works in portrait mode (default one), and does not works in landscape mode (screen autorotate)

same on oneplus 6 (postmarket os, gnome mobile)

so where should i create issue? in telegram desktop repo?

**Comment by YaLTeR** [comment]
Yeah, Telegram desktop should be fine. I noticed they have several output-related problems, like when I have two outputs it will frequently use a wrong scale factor.

**Comment by gmankab** [comment]
https://github.com/telegramdesktop/tdesktop/issues/28847

**Comment by gmankab** [comment]
settings -> advanced -> experimental -> enable precise high dpi scaling

enabling this in telegram desktop fixed bug
